### ISSUE TITLE
FOR loop bugs

### ISSUE THREAD

**Issue by fawltylanguage** [gnudatalanguage/gdl#801]
IDL> for i=32767,32767 do print, i
   32767
IDL>

GDL> for i=32767,32767 do print, i
   32767
  -32768
  -32767
  -32766
  -32765
  ...

----------------------------------------

IDL> for i=20000,30000,10000 do print, i
   20000
   30000
IDL>

GDL> for i=20000,30000,10000 do print, i
   20000
   30000
  -25536
  -15536
   -5536
    4464
   14464
   24464
  -31072
  -21072
  -11072
   -1072
    8928
   18928
   28928
  -26608
  -16608
   -6608
    3392
   13392
   23392
  -32144
  -22144
  -12144
   -2144
    7856
   17856
   27856
  -27680
  -17680
   -7680
    2320
   12320
   22320
GDL>

**Comment by GillesDuvert** [comment]
```
GDL> for i=32767L,32767L do print, i
       32767
GDL> for i=20000LL,30000LL,10000L do print, i
                 20000
                 30000
```

IDL 6.4:
```
IDL> for i=20000,30000,10000 do print, i
   20000
   30000
  -25536
  -15536
   -5536
    4464
   14464
   24464
  -31072
  -21072
  -11072
   -1072
    8928
   18928
   28928
  -26608
  -16608
   -6608
    3392
   13392
   23392
  -32144
  -22144
  -12144
   -2144
    7856
   17856
   27856
  -27680
  -17680
   -7680
    2320
   12320
   22320
IDL>
``` 
I am pleased that IDL after 6.4 (an old refrence that I have around) adopts the ints and not the short ints by default for loops. I personnally never understood why GDL took caution to reproduce the IDL buggy behaviour that IMHO did discourage many potential IDL users.

Shall we say that version 1.0 definitely get rids of this oddity?

**Comment by fawltylanguage** [comment]
16 bit INT works in IDL 8.5.1:

IDL> print, !version                                                                                                                                                                          
{ x86_64 linux unix linux 8.5.1 Nov 14 2015      64      64}                                                                                                                                  
IDL> for i=32767,32767 do help, i                                                                                                                                                             
I               INT       =    32767                                                                                                                                                          
IDL>

**Comment by GillesDuvert** [comment]
replaced by #816

**Comment by GillesDuvert** [comment]
Comment was too hasty: IDL still uses INTs as default unless `COMPILE_OPT DEFINT32` has been issued. (Or `COMPILE_OPT IDL2` ) . This '32767' feature was corrected after 6.4 but other are still present see #816 .
> ```
> GDL> for i=32767L,32767L do print, i
>        32767
> GDL> for i=20000LL,30000LL,10000L do print, i
>                  20000
>                  30000
> ```
> 
> IDL 6.4:
> 
> ```
> IDL> for i=20000,30000,10000 do print, i
>    20000
>    30000
>   -25536
>   -15536
>    -5536
>     4464
>    14464
>    24464
>   -31072
>   -21072
>   -11072
>    -1072
>     8928
>    18928
>    28928
>   -26608
>   -16608
>    -6608
>     3392
>    13392
>    23392
>   -32144
>   -22144
>   -12144
>    -2144
>     7856
>    17856
>    27856
>   -27680
>   -17680
>    -7680
>     2320
>    12320
>    22320
> IDL>
> ```
> 
> I am pleased that IDL after 6.4 (an old refrence that I have around) adopts the ints and not the short ints by default for loops. I personnally never understood why GDL took caution to reproduce the IDL buggy behaviour that IMHO did discourage many potential IDL users.
> 
> Shall we say that version 1.0 definitely get rids of this oddity?
